bin/porta

```diff
@@ -451,29 +451,49 @@ module Porta
     protected
 
     def podman?
-      system('podman --version > /dev/null 2>&1')
+      return @podman if defined? @podman
+
+      @podman = system('podman --version > /dev/null 2>&1')
```

Comment:
Why not?

```suggestion
      @podman ||= system('podman --version > /dev/null 2>&1')
```

Reply:
if it was set, then we never set it again. `||` becomes redundant.

Reply:
I think it's not redundant, it's a simpler way to do the same, like you do in https://github.com/3scale-labs/porta-dev-tools/pull/22/files#diff-422e0549dec116dd1a61191fce586358e3ec8f56ebbe702d29b738ffdbe22a5cR470

Or is it different?

Reply:
it is not the same. If `@podman` is set to `false`, the it will be reevaluated again and again. `||=` will reevaluate whenever the variable is "falsy" which for ruby eans `nil` or `false`. While in my implementation, whenever the variable is defined, it will not be reevaluated anymore.

That's why we also do `hash[:key] = expression unless hash.key?(:key)` instead of `hash[:key] ||= expression` whenever the value could legitimately be `false` or `nil`.

Reply:
Understood